src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,311 @@
+        final List<String> expectedFASTAContents = (null==expectedAssembledContigFile) ? null : Files.readAllLines(Paths.get(expectedAssembledContigFile.getAbsolutePath()));
+
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(actualFASTAContents);
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(expectedFASTAContents);
+        compareContigs(actualAssembledContigs, expectedAssembledContigs);
+    }
+
+    private static void compareContigs(final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs,
+                                       final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedAssembledContigs)
+            throws IOException, InterruptedException{
+
+        final SortedMap<Integer, String> expectedMap = collectContigsByLength(expectedAssembledContigs);
+
+        // first create mapping from read length to sequence
+        // (bad, but contig names are not guaranteed to be reproducible in different runs)
+        final SortedMap<Integer, String> actualMap = collectContigsByLength(actualAssembledContigs);
+
+
+        // essentially, make sure the number of assembled contigs with unique length values are the same
+        final SortedSet<Integer> actualLengthVals = new TreeSet<>(actualMap.keySet());
+        final SortedSet<Integer> expectedLengthVals = new TreeSet<>(expectedMap.keySet());
+        Assert.assertEquals(actualLengthVals.size(), expectedLengthVals.size());
+
+        // first compare contig length values, with some tolerance
+        final Iterator<Integer> itActual = actualLengthVals.iterator();
+        final Iterator<Integer> itExpected = expectedLengthVals.iterator();
+        // if the two sets are not exactly the same but difference is within tolerance, create new map
+        final List<Tuple2<String, String>> actualVSexpectedPairs = new ArrayList<>();
+        while(itActual.hasNext()){
+            final Integer actualLength = itActual.next();
+            final Integer expectedLength = itExpected.next(); // safe operation as the previous assertion guarantees two sets are of same size
+            Assert.assertTrue( Math.abs(actualLength - expectedLength)<=lengthDiffTolerance );
+            actualVSexpectedPairs.add(new Tuple2<>(actualMap.get(actualLength), expectedMap.get(expectedLength)));
+        }
+
+        // then compare sequences, aware of RC
+        for(final Tuple2<String, String> pairOfSimilarLength: actualVSexpectedPairs){
+
+            final String actualString = pairOfSimilarLength._1();
+            final String expectedString = pairOfSimilarLength._2();
+
+            final String rcOfActualString = new String(BaseUtils.simpleReverseComplement(actualString.getBytes(StandardCharsets.UTF_8)));
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.compareContigs\(RunSGAViaProcessBuilderOnSpark$ContigsCollection, RunSGAViaProcessBuilderOnSpark$ContigsCollection\): new String\(byte\[\]\)